TESTS/network/wifi/main.cpp

```diff
@@ -0,0 +1,196 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "utest/utest.h"
+#include "unity/unity.h"
+#include "greentea-client/test_env.h"
+
+#include "mbed.h"
+
+#if TARGET_UBLOX_EVK_ODIN_W2
+#include "OdinWiFiInterface.h"
+#else
+#error [NOT_SUPPORTED] Only built in WiFi modules are supported at this time.
+#endif
+
+using namespace utest::v1;
+
+/**
+ * WiFi tests require following macros to be defined:
+ * - MBED_CONF_APP_WIFI_SSID - SSID of a network the test will try connecting to
+ * - MBED_CONF_APP_WIFI_PASSWORD - Passphrase that will be used to connecting to the network
+ * - WIFI_TEST_NETWORKS - List of network that presence will be asserted e.g. "net1", "net2", "net3"
+ */
+#if !defined(MBED_CONF_APP_WIFI_SSID) || !defined(MBED_CONF_APP_WIFI_PASSWORD) || !defined(MBED_CONF_APP_WIFI_NETWORKS)
+#error MBED_CONF_APP_WIFI_SSID, MBED_CONF_APP_WIFI_PASSWORD and MBED_CONF_APP_WIFI_NETWORKS have to be defined for this test.
```

Comment:
@bulislaw Could you please change this to a `[NOT_SUPPORTED]` macro? This will show up as a build failure for anyone building these tests locally for the UBLOX_EVK_ODIN_W2. If you add the `[NOT_SUPPORTED]` it'll just show up as a skip and they can continue with testing.